pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -377,6 +377,76 @@ func (c *Controller) Instances(hostname model.Hostname, ports []string,
 	return nil, nil
 }
 
+// InstancesByPort implements a service catalog operation
```

Comment:
btw, this is a literal copy of the Instances() function with only one difference. using port number instead of name